model/patron.py

```diff
@@ -300,6 +279,176 @@ def synchronize_annotations(self, value):
                 _db.delete(annotation)
         self._synchronize_annotations = value
 
+    @property
+    def root_lane(self):
+        """Find the Lane, if any, to be used as the Patron's root lane.
+
+        A patron with a root Lane can only access that Lane and the
+        Lanes beneath it. In addition, a patron with a root lane
+        cannot conduct a transaction on a book intended for an older
+        audience than the one defined by their root lane.
+        """
+
+        # Two ways of improving performance by short-circuiting this
+        # logic.
+        if not self.external_type:
+            return None
+        if not self.library.has_root_lanes:
+            return None
+
+        _db = Session.object_session(self)
+        from ..lane import Lane
+        qu = _db.query(Lane).filter(
+            Lane.library==self.library
+        ).filter(
+            Lane.root_for_patron_type.any(self.external_type)
+        )
+        lanes = qu.all()
+        if len(lanes) < 1:
+            # The most common situation -- this patron has no special
+            # root lane.
+            return None
+        if len(lanes) > 1:
+            # Multiple root lanes for a patron indicates a
+            # configuration problem, but we shouldn't make the patron
+            # pay the price -- just pick the first one.
+            logging.error(
+                "Multiple root lanes found for patron type %s.",
+                patron.external_type
+            )
+        return lanes[0]
+
+    def work_is_age_appropriate(self, work_audience, work_target_age):
```

Comment:
Note to myself on Tuesday: This is where I left off reviewing the code and making sure everything has test coverage.

Reply:
Thank you, past self.